R/correct_demographics.R

```diff
@@ -8,23 +8,35 @@
 #' @return episode files with updated date of birth and ages
 #' @export
 correct_demographics <- function(data, year) {
+  # keep episodes with missing chi
+  data_no_chi <- data %>%
+    dplyr::filter(is_missing(.data$chi))
   # Checking and changing DOB and age
-  data <- data %>%
+  data_chi <- data %>%
     dplyr::filter(!is_missing(.data$chi)) %>%
     dplyr::mutate(
       # Create a dob in the previous century from the chi number
       chi_dob_min = phsmethods::dob_from_chi(
         chi_number = .data$chi,
         chi_check = FALSE,
         min_date = lubridate::ymd("1900-01-01"),
-        max_date = pmin(.data$keydate1_dateformat, lubridate::ymd("1999-12-31"), na.rm = TRUE)
+        max_date = pmin(
+          .data$keydate1_dateformat,
+          lubridate::ymd("1999-12-31"),
+          na.rm = TRUE
+        )
       ),
-      # Create a dob in the current century from chi (will return NA if in the future)
+      # Create a DoB in the current century from chi
+      # (will return NA if in the future)
       chi_dob_max = phsmethods::dob_from_chi(
         chi_number = .data$chi,
         chi_check = FALSE,
         min_date = lubridate::ymd("2000-01-01"),
-        max_date = pmax(.data$keydate1_dateformat, lubridate::ymd("2000-01-01"), na.rm = TRUE)
+        max_date = pmax(
+          .data$keydate1_dateformat,
+          lubridate::ymd("2099-12-31"),
+          na.rm = TRUE
+        )
```

Comment:
I don't think this is correct. Here we're looking for a date of birth in the 20thC. The latest that date could be is the end of the year, or the date of the episode (since a person can't be born after an episode has started). So should it be:
```suggestion
        max_date = pmin(
          .data$keydate1_dateformat,
          lubridate::ymd("2099-12-31"),
          na.rm = TRUE
        )
```

Not sure ... 🤯 I'm off home, but I/We maybe need to draw this one out to nail the logic!